### PR TITLE
[WIP] ignore those installations that have not been configured for tracking

### LINKED ISSUE #844: I could not delete wow client in setting
**Describe the bug**
Thx for the awosme ware!
I am a wow-addon author. I have several wow clients in my computer for testing porpuse and I donot want to see some in the updater list.
BUT after I delete some of the clients in the configure tab,  they come back in the next running of wowup.

**To Reproduce**

**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**

**Additional context**

### LINKED ISSUE #990: Toggle switch to ignore updates/notifications for specific version of wow
**Is your feature request related to a problem? Please describe.**
Blueness from discord would like a toggle switch to ignore updates/notifications for specific version of wow in wowup. IE if they are barely playing retail and don't want to be bothered w/ notification about updates for addons for it

**Describe the solution you'd like**
a toggle switch for each version of wow that is installed/detected by wowup to ignore updates/notifications for that specific version of wow

**Describe alternatives you've considered**
Unsure if it would cause issues for other parts of updating/monitoring of the addons

**Additional context** 
screenshot of discord conversation
Screen Shot 2021-07-18 at 8 18 34 PM

### PR DESCRIPTION
**Pull request by jar349**
Closes #844

This PR is the beginning of an implementation for ignoring installations of WoW.

- it expands the model for `WoWInstallation` to include a new property called `defaultTrackAddons`.
- it updates the wow-client-options component to include a slide toggle for `defaultTrackAddons`
- it updates the my-addons component to filter its dropdown based on those installations whose `defaultTrackAddons` is true
- it updates the session service to filter out those installations whose `defaultTrackAddons` is false
- it updates the warcraft-installation service to set the default value of `defaultTrackAddons` on `WoWInstallation` to true

Work still to be done:
- I only speak English, so the other translations need to be filled out
- I've never written TypeScript before, so I have no idea how to test my changes
- Even though the warcraft-installation service creates - by default - instances of `WoWInstallation` where `defaultTrackAddons` is true, and even though the wow-client-options component's slide toggle for tracking addons is set to the state of the `WoWInstallation` model, when the app initially starts, none of the installations are configured to be tracked.  By default, they should ALL be configured to be tracked.  I don't understand why this is happening.

### CODE REVIEW

**Comment by jliddev** [comment]
So the goal of this is essentially to allow users to ignore a WoW installation?

**Comment by jar349** [comment]
Yes, that's my understanding of the desired outcome of #990.

Have I misunderstood?

Do you have any advice for making progress on this PR?

**Comment by jliddev** [comment]
I just wanted to make sure what the goal was before looking it over

**Comment by jliddev** [comment]
This should be for #844 then not #990

**Comment by jar349** [comment]
@jliddev Thank you for that additional issue link. Having read both #844 and #990, it seems to me that what I've begun here will resolve both?  Both of these issues wish to ignore installations of WoW?  Or, are you reading #990 as: "I still want that installation in my dropdown; I just don't want to see the available updates"?

**Comment by jliddev** [comment]
Lets just say #990 is an update notification toggle, and #844 is an ignore completely toggle since they are so similar

**Comment by jar349** [comment]
> Lets just say #990 is an update notification toggle, and #844 is an ignore completely toggle since they are so similar

Acknowledged. I've updated the PR to reflect that it resolved #844.

Regarding the rest of my questions, I could use some help please:
- How shall we get the other translations?
- I could use some hand-holding with testing TypeScript (I assume you require tests?)
- Any idea what I might be doing/assuming incorrectly that causes all installations to begin as toggled off even when I've made the WoWInstallation creation service default to `true`?

**Comment by jliddev** [comment]
**How shall we get the other translations?**

You can use "npm run i18n" to fill in the other language templates and the translators can jump in then.

**I assume you require tests?**

There are test examples all over in the .spec files, however wildly incomplete due to time and laziness.

File: wowup-electron/src/app/components/wow-client-options/wow-client-options.component.html [hunk 1]
```diff
@@ -50,6 +50,15 @@
         (change)="onDefaultAutoUpdateChange($event)">
       </mat-slide-toggle>
     </div>
+    <div class="row">
```
Comment (jliddev) [inline wowup-electron/src/app/components/wow-client-options/wow-client-options.component.html:53]: Lets maybe move this up to be the first option under the file path just so it seems important

File: wowup-electron/src/app/models/wowup/wow-installation.ts [hunk 2]
```diff
@@ -9,4 +9,5 @@ export interface WowInstallation {
   selected: boolean;
   defaultAddonChannelType: AddonChannelType;
   defaultAutoUpdate: boolean;
+  defaultTrackAddons: boolean;
```
Comment (jliddev) [inline wowup-electron/src/app/models/wowup/wow-installation.ts:12]: Per ticket discussion lets just change this to ```enabled: boolean;```

File: wowup-electron/src/app/components/wow-client-options/wow-client-options.component.ts [hunk 3]
```diff
@@ -145,6 +145,14 @@ export class WowClientOptionsComponent implements OnInit, OnDestroy {
     this.installationModel.defaultAutoUpdate = evt.checked;
   }
 
+  public onDefaultTrackAddonsChange(evt: MatSlideToggleChange): void {
```
Comment (jliddev) [inline wowup-electron/src/app/components/wow-client-options/wow-client-options.component.ts:148]: Rename to reflect new property name ```onEnabledChange(```

File: wowup-electron/src/app/pages/my-addons/my-addons.component.html [hunk 4]
```diff
@@ -9,7 +9,7 @@
         <mat-label>{{ "PAGES.MY_ADDONS.CLIENT_TYPE_SELECT_LABEL" | translate }}</mat-label>
         <mat-select class="select pointer" [(value)]="selectedInstallationId" (selectionChange)="onClientChange()"
           [disabled]="enableControls === false" [disableOptionCentering]="true">
-          <mat-option [value]="installation.id" *ngFor="let installation of wowInstallations$ | async">
+          <mat-option [value]="installation.id" *ngFor="let installation of trackingEnabled(wowInstallations$) | async">
```
Comment (jliddev) [inline wowup-electron/src/app/pages/my-addons/my-addons.component.html:12]: Probably dont want to do this, probably want to edit the  ```WarcraftInstallationService``` to provide a new observable based on the current ```wowInstallations$``` filtered to only enabled ones, something like ```enabledWowInstallation$```

File: wowup-electron/src/app/pages/my-addons/my-addons.component.ts [hunk 5]
```diff
@@ -882,6 +882,13 @@ export class MyAddonsComponent implements OnInit, OnDestroy, AfterViewInit {
     this.gridApi?.deselectAll();
   }
 
+  public trackingEnabled(installations: Observable<WowInstallation[]>): Observable<WowInstallation[]> {
```
Comment (jliddev) [inline wowup-electron/src/app/pages/my-addons/my-addons.component.ts:885]: See above comment

File: wowup-electron/src/assets/i18n/en.json [hunk 6]
```diff
@@ -478,7 +478,9 @@
         "RESCAN_CLIENTS_BUTTON": "Re-Scan",
         "RESCAN_CLIENTS_LABEL": "Rescan installed World of Warcraft products",
         "SAVE_WOW_DIRECTORY_SELECT_BUTTON": "Save",
-        "TITLE": "World of Warcraft"
+        "TITLE": "World of Warcraft",
+        "TRACK_ADDONS_LABEL": "Track Addons",
```
Comment (jliddev) [inline wowup-electron/src/assets/i18n/en.json:482]: change to something like ```CLIENT_ENABLED_LABEL``` with text ```Enabled```

File: wowup-electron/src/assets/i18n/en.json [hunk 7]
```diff
@@ -478,7 +478,9 @@
...
+        "TRACK_ADDONS_DESCRIPTION": "If enabled, this application's addons will be tracked"
```
Comment (jliddev) [inline wowup-electron/src/assets/i18n/en.json:483]: Change to ```CLIENT_ENABLED_DESCRIPTION``` with text ```If disabled, this WoW client and all addons will be ignored```

**Review by jliddev — COMMENTED** [review]
Needs an extra step to migrate existing installations forward.
```WarcraftInstallationService line 177``` add some new step thats like ```this.migrateInstallation``` that will check to see if the new required property exists on the installation and if not set it to the default true

**Comment by jar349** [comment]
Thank you for the very helpful review. I will make these changes soon.

**Pull request closed without merging**